relnotes/taskext.feature.md

```diff
@@ -0,0 +1,70 @@
+## Built-in task and scheduler support in DaemonApp
+
+`ocean.util.app.DaemonApp` `ocean.util.app.ext.TaskExt`
+
+It is now possible to make `DaemonApp` initialize scheduler automatically using
+standard app configuration file. To do so, it is sufficient to enable it via
+`DaemonApp` optional settings:
+
+```D
+class MyApp : DaemonApp
+{
+  this ( )
+  {
+    DaemonApp.OptionalSettings settings;
+    settings.use_task_ext = true;
+    super("name", "desc", VersionInfo.init, settings);
+  }
+```
+
+This extension is disabled by default to avoid clashing with existing
+application code which is already handling scheduler initialization in some way.
+
+If `TaskExt` is enabled, overriden `run` method will be called within a task
+context and there is no need to create one manually or start scheduler event
+loop:
+
+```D
+class MyApp : DaemonApp
+{
+  this ( )
+  {
+    DaemonApp.OptionalSettings settings;
+    settings.use_task_ext = true;
+    super("name", "desc", VersionInfo.init, settings);
+  }
+
+  override int run ( Arguments args, ConfigParser config )
+  {
+    assert(Task.getThis() !is null);
+    assert(isSchedulerUsed());
+
+    scope (exit)
+    {
+      // must be called at some point, registered DaemonApp events won't let
+      // event loop exit on its own:
+      theScheduler.epoll.shutdown();
+      // theScheduler.shutdown(); // alternative to also hard kill
```

Comment:
Does the use of these `scope (exit)` options assume that at some point in the `run` method, there will be some kind of epoll-using thing that happens that blocks until the app is done?

I'm thinking of the use-case of a `DaemonApp`, where the `run` method might spin off a number of things that repeatedly schedule various epoll events, but which doesn't do anything that _blocks_.  In that case one might exit the run method and therefore trigger the `shutdown` call, when actually one wants the eventloop to keep running.

Reply:
Any `DaemonApp` has such epoll using things from timer ext and periodic stats log. If you want to keep everything running (and have some own condition for terminating things), you can simply remove shutdown line - in that case main task will exit but everything else will keep running.

Reply:
Fair enough, but it might be worth considering whether the release notes could be tweaked a bit.  This documented example might be a bit misleading, as I suspect quite a lot of practical application uses don't have anything that blocks long-term -- they just spin up a bunch of different things that keep triggering new tasks and/or epoll events.

So, I'm not sure that `scope (exit) theScheduler.epoll.shutdown();` is advisable in many circumstances, although it's obviously useful to have it listed as a possible option.

Reply:
Example was only to show difference with old `run` method which would end simultaneously with the end of an app. Any phrasing suggestions for release notes?

Reply:
I'll try to think of something, and if I come up with an idea, submit a PR.